plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeCommitSummary.java

```diff
@@ -59,15 +61,16 @@ else if (transactionLogEntry.getAdd() != null) {
                 addedFilesCanonicalPartitionValuesBuilder.add(transactionLogEntry.getAdd().getCanonicalPartitionValues());
             }
             else if (transactionLogEntry.getRemove() != null) {
-                removedFilesFound = true;
+                Map<String, String> partitionValues = transactionLogEntry.getRemove().partitionValues();
+                removedFilesCanonicalPartitionValuesBuilder.add(partitionValues == null ? ImmutableMap.of() : canonicalizePartitionValues(partitionValues));
```

Comment:
Why do we add empty map to set?

Reply:
Good catch.
In the unlikely case that the winning commit does not contain information about the partition values of the `remove` file, we should fail the operation

Reply:
Should we also fix addFiles in the same way?